### ISSUE TITLE
Extend support for user secrets and appsettings to use .jsonc

### ISSUE THREAD

**Issue by Nairda015** [dotnet/aspnetcore#49583]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the solution you'd like

Hey 
There is an issue about this already but it is closed and I think approached in the wrong way. https://github.com/dotnet/aspnetcore/issues/13070

Instead of changing from JSON to JSONC, I propose to extend the supported extensions list for all appsettings and usersecrets.
I know JSON with comments works so the only reason to add this is to remove red underlines.

<img width="861" alt="image" src="https://github.com/dotnet/aspnetcore/assets/44712992/af8cea9d-f552-4dc8-ab02-2aa66a4326e1">

**Comment by mitchdenny** [comment]
I don't think the juice is worth the squeeze on this one. Adding support for ```.jsonc``` extensions in addition to ```.json``` would lead to a fair amount of confusion in terms of which one takes precedence if the other exists (do we take one or the other, or do we merge them, and what happens when there are conflicts, and what happens if the result of the merge is invalid configuration).

I'm not sure which IDE you are using (Rider?). I'm surprised it doesn't at least give you the option to suppress these warnings?

**Comment by Nairda015** [comment]
I don't think that using/merging appsettings.json with appsettings.jsonc is a good idea but applying appsettings.Dev.jsonc on top of appsettigns.json don't raise any concerns in my opinion

**Comment by mitchdenny** [comment]
Having comments in JSON files is a very common practice. Doing an overhaul of the configuration system to allow people to use *.jsonc files instead of *.json files doesn't seem worth it to me. But we can put this on the backlog and see if we get more feedback about it.

**Comment by mitchdenny** [comment]
Actually, upon further reflection this issue is filed in the wrong place. Whilst ASP.NET Core does use the configuration system; this is really controlled by the config APIs and host model. So you might like to follow up in dotnet/runtime. I'll transfer the issue there for you.
